External/FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp

```diff
@@ -5303,6 +5303,46 @@ void OpDispatchBuilder::MOVBEOp(OpcodeArgs) {
   StoreResult(GPRClass, Op, Src, 1);
 }
 
+template<size_t ElementSize>
+void OpDispatchBuilder::HADDP(OpcodeArgs) {
+  auto Size = GetSrcSize(Op);
+  OrderedNode *Dest = LoadSource(FPRClass, Op, Op->Dest, Op->Flags, -1);
+  OrderedNode *Src = LoadSource(FPRClass, Op, Op->Src[0], Op->Flags, -1);
+
+  OrderedNode *Res = _VFAddP(Size, ElementSize, Dest, Src);
+  StoreResult(FPRClass, Op, Res, -1);
+}
+
+template<size_t ElementSize>
+void OpDispatchBuilder::HSUBP(OpcodeArgs) {
+  auto Size = GetSrcSize(Op);
+  OrderedNode *Dest = LoadSource(FPRClass, Op, Op->Dest, Op->Flags, -1);
+  OrderedNode *Src = LoadSource(FPRClass, Op, Op->Src[0], Op->Flags, -1);
+
+  // This is a bit complicated since AArch64 doesn't support a pairwise subtract
```

Comment:
I hope this leads to some good optimisation opportunities.

Reply:
I hope so as well :)